migration-guide.md

```diff
@@ -0,0 +1,317 @@
+# Migration to Plotly 3.0.0
+There are many new and great features in Plotly 3.0 including deeper Jupyter integration, deeper figure validation, improved performance, and more. To get started right away with Plotly, check out the [tutorial](#getting-started) below.
+
+## 3.0.0rc9 [08-06-2018]
+### Added
+- Jupyter integration - run `help(go.FigureWidget)` for more information
+- update traces interactively
+- Traces can be added and updated interactively by simply assigning to properties
+- The full Traces and Layout API is generated from the plotly schema to provide a great experience for interactive use in the notebook
+- Support for setting array properties as numpy arrays. When numpy arrays are used, ipywidgets binary serialization protocol is used to avoid converting these to JSON strings.
+- Context manager API for animation. Run `help(go.Figure().batch_animate)` for the full doc string.
+- Programmatic export of figures to static SVG images (and PNG and PDF with cairosvg installed).
+
+### Removed
+- Removed `.to_string`, `.strip_style`, `.get_data`, `.validate` and `.to_dataframe` methods from `plotly.graph_objs` objects. For example run `dir(plotly.graph_objs.Scatter)` to get all the (magic) methods of the Scatter class.
+
+
+### Changed
+- Improved data validation covering the full API with clear, informative error messages. This means that incorrect properties and/or values now always raise a `ValueError` with a description of the error, the invalid property, and the avaialble properties on the level that it was placed in the graph object. Eg. `go.Scatter(foo=123)` raises a validation error. See https://plot.ly/python/reference/ for a reference to all valid properties and values in the Python API.
+
+- graph objects are not `dict`s anymore. Running a cell of a graph object prints out a dict-style representation of the object:
+
+Eg. `plotly.graph_objs.Scatter()` prints
+
+```
+Scatter(**{
+    'type': 'scatter'
+})
+```
+
+- plotly objects now have a `.to_plotly_json` method that converts the object to a dict:
```

Comment:
plotly objects now -> plotly objects still

This method was already present for graph_objs right?

Reply:
I don't believe it was